addons/project/models/project.py

```diff
@@ -2061,7 +2061,11 @@ def write(self, vals):
 
         now = fields.Datetime.now()
         if 'parent_id' in vals and vals['parent_id'] in self.ids:
-            raise UserError(_("Sorry. You can't set a task as its parent task."))
+            for task in self:
+                if task in self.child_ids:
+                    task.parent_id = False
+                if vals['parent_id'] == task.id:
+                    self = self - task
```

Comment:
I'm discussing it with aju, but to me, the bug is that the error should be raised and isn't

Reply:
Yes @xavierbol also agrees. What should be fixed is the fact that the error isn't showing.

Reply:
Hello @auon-odoo Thank you for looking into it. :slightly_smiling_face: 

I think this is a generic issue that occurs when editing records in batch mode. 
In such cases, a confirmation dialog box with "Confirm" and "Cancel" buttons is typically displayed. The "Confirm" button triggers a call to the ORM write method. However, if an error occurs within the write method, the catch block fails to display the error and instead discards the dialogbox. [Link](https://github.com/odoo/odoo/blob/16.0/addons/hr_timesheet/models/hr_timesheet.py#L290C1-L296C100) 

What do you think? :slightly_smiling_face:

Reply:
@yava-odoo I think it deserves a fix :)

Reply:
Hello @auon-odoo 
I've implemented the changes. Thank you :slightly_smiling_face: